src/transforms/filter.jl

```diff
@@ -0,0 +1,60 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(function)
+
+Filters the table returning only the rows where the `function` returns true.
+"""
+struct Filter{F} <: Stateless
+  func::F 
+end
+
+isrevertible(::Type{<:Filter}) = true
+
+rowvalues(row, colnames) = [Tables.getcolumn(row, col) for col in colnames]
+
+function apply(transform::Filter, table)
+  tablerows = Tables.rows(table)
+  colnames = Tables.columnnames(table)
+  rows = [rowvalues(row, colnames) for row in tablerows]
```

Comment:
If we need to collect the rows anyways, then we can simply use the Tables.rowtable helper function.

Reply:
OK, I'll do it.